6_Agent_Deployment/backend_agent_api/agent_api.py

```
@@ -493,6 +493,133 @@
+        ]:
+            # TODO: Implement proper Excel parsing (e.g., pandas/openpyxl) and row extraction
+            # For now, treat as generic binary file
+            file_content = f"Binary spreadsheet: {request.file_name} ({len(file_bytes)} bytes)"
+        
+        # Chunk the text
+        chunks = chunk_text(file_content)
+        
+        # Generate embeddings for all chunks in one call
+        embeddings = create_embeddings(chunks) if chunks else []
+        if chunks and len(embeddings) != len(chunks):
+            raise HTTPException(status_code=500, detail="Embedding count mismatch")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)